src/NzbDrone.Core/Indexers/Tribler/TriblerIndexer.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http.Headers;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Http;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.Exceptions;
+using NzbDrone.Core.Http.CloudFlare;
+using NzbDrone.Core.Indexers.Exceptions;
+using NzbDrone.Core.IndexerSearch.Definitions;
+using NzbDrone.Core.Parser;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.Validation;
+using Tribler.Api;
+
+namespace NzbDrone.Core.Indexers.Tribler
+{
+    public class TriblerIndexer : HttpIndexerBase<TriblerIndexerSettings>
+    {
+        public override string Name => "Tribler";
+
+        public override DownloadProtocol Protocol => DownloadProtocol.Torrent;
+
+        // TODO: Consider using channels as rss feed.
+        public override bool SupportsRss => false;
+        public override bool SupportsSearch => true;
+
+        public override int PageSize => 50;
+
+        // this is not a public service so we do no not have to handle rate limitations.
+        public override TimeSpan RateLimit => TimeSpan.FromSeconds(0);
+
+        public TriblerIndexer(IHttpClient httpClient, IIndexerStatusService indexerStatusService, IConfigService configService, IParsingService parsingService, Logger logger)
+            : base(httpClient, indexerStatusService, configService, parsingService, logger)
+        {
+        }
+
+        public override IIndexerRequestGenerator GetRequestGenerator()
+        {
+            return new TriblerIndexerRequestGenerator() { Settings = Settings };
+        }
+
+        public override IParseIndexerResponse GetParser()
+        {
+            return new TriblerIndexerResponseParser(Settings);
+        }
+
+        public override IList<ReleaseInfo> FetchRecent()
+        {
+            return new List<ReleaseInfo>();
+        }
+
+        // override the default Test command as it requires support for rss feed.
+        protected override void Test(List<ValidationFailure> failures)
+        {
+            NzbDroneValidationResult nzbDroneValidationResult = Settings.Validate();
+            failures.AddRange(nzbDroneValidationResult.Errors);
+
+            try
+            {
+                // fake a search for "Ubuntu"
+                var requestGenerator = (TriblerIndexerRequestGenerator)GetRequestGenerator();
+
+                var indexerRequest = requestGenerator.GetRequest("Ubuntu").First();
```

Comment:
For GetRecent you seem to be searching for "*", why not do the same here

Reply:
That GetRequest("*") is actully never used because the FetchRecent is overridden in the indexer.
And it's incorrect as well. I will add a exception instead.